### ISSUE TITLE
Create shared library for multiple microservices failed on Linux platform

### ISSUE THREAD

**Issue by rushwing** [Nuitka/Nuitka#2294]
**Problem**: We have multiple microservices written in Python, and shared some third-party packages.
We could create standalone executables for different microservices, but the size might be huge, compared with using shared library. So we plan to create a shared library which would include all dependent third-party packages with the following command:
> python3.10 -m nuitka --module shared-env/lib/python3.10/site-packages/ --include-package grpcio --include-package grpcio_tools --include-package protobuf --include-package fastapi --include-package pydantic --include-package starlette --include-package uvicorn --include-package requests --include-package pandas --include-package pika --include-package click --include-package mypy_extensions --include-package nuitka --include-package packaging --include-package pathspec --include-package platformdirs --standalone shared.py

, and then build each of the microservices using the following command:
> python3.10 -m nuitka --include-module shared.so --standalone service_a.py

But when we built the shared.so, it report the following error message:
> FATAL: 
Error, specify only one positional argument unless "--run" is specified to
pass them to the compiled program execution.

We follow the instructions suggested by ChatGPT to modify the command of generating shared library as follows:
> python3.10 -m nuitka --module --run shared-env/lib/python3.10/site-packages/ --include-package grpcio --include-package grpcio_tools --include-package protobuf --include-package fastapi --include-package pydantic --include-package starlette --include-package uvicorn --include-package requests --include-package pandas --include-package pika --include-package click --include-package mypy_extensions --include-package nuitka --include-package packaging --include-package pathspec --include-package platformdirs --standalone shared.py

Then it reported error:
> Nuitka-Options:WARNING: You did not specify to follow or include anything but main module. Check options and make sure that is intended.
> Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.10/3.10.12/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/homebrew/Cellar/python@3.10/3.10.12/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/xxxx/xxxx/shared-env/lib/python3.10/site-packages/nuitka/tools/data_composer/__main__.py", line 40, in <module>
    main()
  File "/Users/xxxx/xxxx/shared-env/lib/python3.10/site-packages/nuitka/tools/data_composer/DataComposer.py", line 412, in main
    assert name not in names, name
AssertionError
Traceback (most recent call last):
  File "/Users/xxxx/xxxx/shared-env/lib/python3.10/site-packages/nuitka/__main__.py", line 155, in <module>
    main()
  File "/Users/xxxx/xxxx/shared-env/lib/python3.10/site-packages/nuitka/__main__.py", line 141, in main
    MainControl.main()
  File "/Users/xxxx/xxxx/shared-env/lib/python3.10/site-packages/nuitka/MainControl.py", line 1103, in main
    _main()
  File "/Users/xxxx/xxxx/shared-env/lib/python3.10/site-packages/nuitka/MainControl.py", line 972, in _main
    result, options = compileTree()
  File "/Users/xxxx/xxxx/shared-env/lib/python3.10/site-packages/nuitka/MainControl.py", line 882, in compileTree
    runDataComposer(source_dir)
  File "/Users/xxxx/xxxx/shared-env/lib/python3.10/site-packages/nuitka/build/DataComposerInterface.py", line 33, in runDataComposer
    blob_filename = _runDataComposer(source_dir=source_dir)
  File "/Users/xxxx/xxxx/shared-env/lib/python3.10/site-packages/nuitka/build/DataComposerInterface.py", line 54, in _runDataComposer
    subprocess.check_call(
  File "/opt/homebrew/Cellar/python@3.10/3.10.12/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/Users/xxxx/xxxx/shared-env/bin/python3.10', '/Users/xxxx/xxxx/shared-env/lib/python3.10/site-packages/nuitka/tools/data_composer', 'artifacts/.build', 'artifacts/.build/__constants.bin']' returned non-zero exit status 1.

* Nuitka version, full Python version, flavor, OS, etc. as output by *this*
  command.

  > 1.6.3
Commercial: None
Python: 3.10.12 (main, Jun  7 2023, 00:38:32) [Clang 14.0.3 (clang-1403.0.22.14.1)]
Flavor: Homebrew Python
Executable: /Users/danielwong/PycharmProjects/OculusManufacturing/arvr/projects/manufacturing/columbus/columbus_app/columbus_app_venv/bin/python3.10
OS: Darwin
Arch: arm64
Version C compiler: /usr/bin/clang (clang).

* How did you install Nuitka and Python

  > I installed Python in a virtualenv, and install Nuitka with `python3.10 -m pip install -r pip_requirements.txt`

* The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g.
  `numpy`, you are expected to shorten this to the relevant ones.

  > python -m pip freeze

* Provide in your issue the Nuitka options used

> --module --run shared-env/lib/python3.10/site-packages/ --include-package grpcio --include-package grpcio_tools --include-package protobuf --include-package fastapi --include-package pydantic --include-package starlette --include-package uvicorn --include-package requests --include-package pandas --include-package pika --include-package click --include-package mypy_extensions --include-package nuitka --include-package packaging --include-package pathspec --include-package platformdirs --standalone shared.py

**Comment by kayhayen** [comment]
Please don't waste my time with AI phantasies. Instead, read the docs. This amount of included packages makes no sense, esp. including Nuitka itself is funny. And in module mode, you only include your own code, many of the packages you listed will not work or coexist in any way. And then using site-packages for the module code is outright funny.

This is a very good illustration, on why AI bot destroy things for good people. But I think I will use you for some very nice reflection on how bad things can get.

As for size issues, you may want to read up on multidist, which allows to deploy multiple programs from one binary.
